### DISCUSSION TITLE
RT3S I/O error

### DISCUSSION THREAD

**Discussion by HakanKoseoglu** [hmatuschek/qdmr#361] [General]
Hi all,

I got a pair of 2nd hand RT3S devices. 

I compiled the QDMR from source on an Ubuntu 23.04, and blacklisted the drivers successfully. The radio is successfully detected, and I could download the existing data. After modifying my callsign, DMR ID and the welcome screen strings, I attempted to uplod back into the radio, and encountered the following, looks like radio does a reset in the middle of the upload but I'm not sure if this is intentional.

Debug in lib/tyt_interface.cc@56: Found device TyT MD-UV390 at USB device in DFU mode: bus 1, device 22.
Debug in lib/tyt_radio.cc@218: Encode codeplug.
Debug in lib/tyt_radio.cc@226: Upload 2 elements.
ERROR in lib/dfu_libusb.cc@239: Cannot get status: Input/Output Error.
ERROR in lib/dfu_libusb.cc@201: Cannot write to device: No such device (it may have been disconnected).
ERROR in lib/dfu_libusb.cc@432: Cannot leave DFU mode.
Debug in src/application.cc@736: Write complete.
Debug in lib/tyt_radio.cc@26: Destructed TyT radio.

As a result no menu buttons are available, the radio simply lists Unprogrammed 1 and not capable of doing anything else.

Any help is welcome!

**Comment by hmatuschek** [comment]
The line 
```
 ERROR in lib/dfu_libusb.cc@201: Cannot write to device: No such device (it may have been disconnected).
```
indicates, that the connection to the computer is reset. This can happen, if the cable is broken and there is an intermittent connection to the device. Consequently a incomplete codeplug is written to the device producing a lot of issues. Something you could try:

1.  If you have several devices, try using a different cable. 
2. Some radios (I do not know if the RT3S suffers from that) the audio and USB share the same lines. Then, you have to turn the volume down to a minimum to get a reliable connection to the radio. 
3. The battery should not be drained when programming. I had some error reports on that. 

Hope, that some of the hints help.

**Reply by HakanKoseoglu** [reply]
Cable is OK, I managed to reset the device and then clone from the spare using Retevis's Windows utility using the same.

I will try the volume button trick. Thanks!
